libs/application/ui-fields/src/lib/TableRepeaterFormField/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { TableRepeaterItem } from '@island.is/application/types'
+
+type Item = {
+  id: string
+} & TableRepeaterItem
+
+export const checkForCustomMappedComponents = (
+  items: Array<Item>,
+  values: any,
+) => {
+  items.some((item) => {
+    if (item.component === 'nationalIdWithName') {
+      handleNationalIdWithName(item, values)
+    }
+  })
+}
+
+const handleNationalIdWithName = (item: Item, values: any) => {
+  // nationalIdWithName returns an object that we
+  // need to extract entries from and add to values
+  if (values) {
+    values.forEach((value: any, index: string) => {
+      if (typeof value[item.id] === 'object') {
+        const { [item.id]: nestedObject, ...rest } = value
+        values[index] = { ...nestedObject, ...rest }
+      }
+    })
+  }
+}
```

Comment:
Can this be done with a type or a generic instead of any?
This isn't returning anything, wouldn't it be better to return a new object instead of changing the input object